nix/neovim-overlay.nix

```diff
@@ -84,12 +84,20 @@ with final.pkgs.lib; let
     lua-language-server
     nil # nix LSP
   ];
+
+  # These are runtime lua dependencies.
+  extraLuaPackages = ps:
+    with ps; [
+      # LuaSnip dependency.
+      jsregexp
+    ];
```

Comment:
issue: this should be unnecessary.

luasnip [exists as a lua package](https://github.com/NixOS/nixpkgs/blob/1042fd8b148a9105f3c0aca3a6177fd1d9360ba5/pkgs/development/lua-modules/generated-packages.nix#L2344), and already has the `jsregexp` dependency defined.

`buildNeovimPlugin` (used to build neovim plugins in nixpkgs) searches the lua package set for a package of the same name and, if found, creates a neovim plugin from it, which includes all dependencies.

There has been a bug in which the dependencies were not propagated, but I've been told that it has been fixed. If it's not in nixpkgs-unstable yet, it will be soon.

The way I recommend to resolve plugin dependencies in nixpkgs is:

- Ask the plugin author(s) to publish their plugins to luarocks (see [this guide](https://github.com/nvim-neorocks/sample-luarocks-plugin)).
- Add the plugin(s) to the nixpkgs lua package set (or open a packaging request in nixpkgs).

That way, declaring dependencies becomes the plugin author's responsibility, not the user's.

Reply:
Great I didn't know about that! It will simplify a lot of my own neovim overlay. :smile: 

So do you think that we could put an example of `extraLuaPackages` to guide future users? Otherwise I can just remove this part from the overlay.

Reply:
Also, I have put the `extraLuaPackages` above the comment `# The below arguments can typically be left as their defaults` because it is used in the overlay, but based on your comment it should not be used that often so maybe I can put it below that comment line again.

Reply:
> So do you think that we could put an example of extraLuaPackages to guide future users?

I would just put an example in the comment in `mkNeovim`.

> so maybe I can put it below that comment line again.

Yes :smile:

Reply:
This should be resolved! You can have a look at the extended comment in `mkNeovim`. I kept the example with `jsregexp` because at least this is something that does not exist outside the luaPackages so it should be clear for a user that reads the doc where the packages are coming from. I can put something else if you want (do not hesitate!) or make a final rebase before merge.